core/arch/x86/mangle.c

```diff
@@ -1823,6 +1826,7 @@ mangle_return(dcontext_t *dcontext, instrlist_t *ilist, instr_t *instr,
         IF_X64(ASSERT_TRUNCATE(val, int, opnd_get_immed_int(instr_get_src(instr, 0))));
         /* addl sizeof_param_area, %xsp
          * except that clobbers the flags, so we use lea */
+        /* XXX i#3307: necessary to instr_set_translation_mangling_epilogue? */
```

Comment:
Looking at translate_walk_track(), this OP_lea won't match any of the cases, and so will set unsupported_mangle to true and just fail, which is a perfectly acceptable thing to do for a situation that can't involve a synchronous fault.

In fact this brings up an alternative way to solve this rip-rel case: if it's easy to identify the spilled reg restore for rip-rel we can just fail and not bother to translate.  Not sure we were seriously considering that before?  Is that simpler than doing all of this?

Reply:
I did not consider this as I thought it's a performance hit or creates other issues.

Reply:
Offline discussion.